server/src/components/Deliveries.js

```diff
@@ -0,0 +1,198 @@
+    this.isPinned = this.isPinned.bind(this)
+  }
+
+  componentDidMount () {
+    this.setupEventSource()
+  }
+
+  setupEventSource () {
+    const url = window.location.pathname
+    console.log('Connecting to event source:', url)
+    this.events = new window.EventSource(url)
+    this.events.onopen = this.onopen.bind(this)
+    this.events.onmessage = this.onmessage.bind(this)
+    this.events.onerror = this.onerror.bind(this)
+  }
+
+  onopen (data) {
+    this.setState({
+      connection: true
+    })
+  }
+
+  onerror (err) {
+    this.setState({
+      connection: false
+    })
+    switch (this.events.readyState) {
+      case window.EventSource.CONNECTING:
+        console.log('Reconnecting...', err)
+        break
+      case window.EventSource.CLOSED:
+        console.log('Reinitializing...', err)
+        this.setupEventSource()
+        break
+    }
+  }
+
+  onmessage (message) {
+    console.log('received message!')
+    const json = JSON.parse(message.data)
+
+    // Prevent duplicates in the case of redelivered payloads
+    const idProp = 'x-github-delivery'
+    if (this.state.log.findIndex(l => l[idProp] === json[idProp]) === -1) {
+      this.setState({
+        log: [json, ...this.state.log]
+      }, () => {
+        localStorage.setItem(this.ref, JSON.stringify(this.state.log.slice(0, this.storageLimit)))
+      })
+    }
+  }
+
+  clear () {
+    if (confirm('Are you sure you want to clear the delivery log?')) {
+      console.log('Clearing logs')
+      const filtered = this.state.log.filter(this.isPinned)
+      this.setState({ log: filtered })
+      if (filtered.length > 0) {
+        localStorage.setItem(this.ref, JSON.stringify(filtered))
+      } else {
+        localStorage.removeItem(this.ref)
+      }
+    }
+  }
+
+  togglePinned (id) {
+    const deliveryId = this.state.pinnedDeliveries.indexOf(id)
+    let pinnedDeliveries
+    if (deliveryId > -1) {
+      pinnedDeliveries = [
+        ...this.state.pinnedDeliveries.slice(0, deliveryId),
+        ...this.state.pinnedDeliveries.slice(deliveryId + 1)
+      ]
+    } else {
+      pinnedDeliveries = [...this.state.pinnedDeliveries, id]
+    }
+
+    this.setState({ pinnedDeliveries })
+    localStorage.setItem(this.pinnedRef, JSON.stringify(pinnedDeliveries))
+  }
+
+  isPinned (item) {
+    const id = item['x-github-delivery']
+    return this.state.pinnedDeliveries.includes(id)
+  }
+
+  render () {
+    const { log, filter, pinnedDeliveries } = this.state
+    let filtered = log
+    if (filter) {
+      filtered = log.filter(l => {
+        if (filter && filter.includes(':')) {
+          let [searchString, value] = filter.split(':')
+          if (!searchString.startsWith('body')) searchString = `body.${searchString}`
+          console.log(l, searchString, value)
+          return get(l, searchString) === value
+        }
+        return true
+      })
+    }
+    const handlers = {
+      'moveUp': (e) => console.log('pressed up'),
```

Comment:
I realize you're still working on the handlers, but these object keys don't need the single-quotes

Reply:
Yeh this is how they had it in their documentation. Seemed weird, but it worked. Will adjust